### PR TITLE
always use test-specific temporary data dir for Python tests (#822)

### PR DESCRIPTION
**Pull request by laneb**
Fixing #822.

### CODE REVIEW

**Comment by laneb** [comment]
Need to fix something. Will reopen when ready for review.

File: src/main/python/test_support/smvbasetest.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import os, shutil
 
 class SmvBaseTest(unittest.TestCase):
+    """DataDir value is deprecated. Use tmpDataDir instead."""
```
Comment (AliTajeldin) [inline src/main/python/test_support/smvbasetest.py]: The above comments makes the warning the doc string for SmvBaseTest.  Just use the `#` instead

File: src/main/python/test_support/smvbasetest.py [hunk 2]
```diff
@@ -82,14 +83,22 @@ def tmpTestDir(cls):
         return cls.PytestDir + "/" + cls.__name__
 
     @classmethod
+    def tmpDataDir(cls):
```
Comment (AliTajeldin) [inline src/main/python/test_support/smvbasetest.py]: Add comment about what this is for and why we need it (as opposed to good old datadir)

File: src/main/python/test_support/smvbasetest.py (hunk 1 above)
Comment (laneb) [inline src/main/python/test_support/smvbasetest.py]: Oh good catch. I meant for it to be the docstring for the attribute but that probably doesn't work if it's the first line of the class. I guess we're not documenting the `SmvBaseTest` API yet anyway.

File: src/main/python/test_support/smvbasetest.py (hunk 2 above)
Comment (laneb) [inline src/main/python/test_support/smvbasetest.py]: Fixed.

File: src/main/python/test_support/smvbasetest.py (hunk 1 above)
Comment (laneb) [inline src/main/python/test_support/smvbasetest.py]: Fixed.

**Pull request merged**